backend/models/user.model.ts

```diff
@@ -62,4 +64,16 @@ const UserSchema: Schema = new Schema({
   },
 });
 
+UserSchema.post("findOneAndDelete", async (doc) => {
+  if (doc.role !== "Teacher") return;
+
+  /* eslint-disable no-underscore-dangle */
+  await MgTestSession.deleteMany({ teacher: doc._id });
+  await MgSchool.findOneAndUpdate(
+    { teachers: doc._id },
+    { $pull: { teachers: doc._id } },
+    { new: true },
+  );
+});
```

Comment:
bit of a nit: but should we use `pre` hook here instead of `post`? i.e delete child dependencies first and then the parent?

Reply:
hmm that makes sense - but i was also thinking if user deletion fails, would we need to roll back the changes?

Reply:
LOL I was actually also thinking about that when I added this comment, i.e if deletion fails, is it better to have undeleted test sessions or undeleted teacher? 👀

i.e if you deleted the test session and it worked, but then the teacher didn't delete, it would still be fine (b/c teacher would just have one less test session). but if deleted teacher & the test session didn't delete, then there would be a test session w/ an undefined teacher field... and not sure if we handle that as gracefully? but maybe our code should just handle all those cases gracefully? idk

TLDR i'm not actually sure what is best here 😢

Reply:
i think that makes sense. so we delete the dependencies before the teacher. i can look into rolling back the changes for that too